packages/client/src/WalletAPIClient.ts

```diff
@@ -82,9 +85,25 @@ export class WalletAPIClient extends RpcNode<
    */
   public exchange: ExchangeModule;
 
+  /**
+   * Instance of the Custom module
+   */
+  public custom: CustomGetter extends (client: WalletAPIClient<M>) => infer U
+    ? U
+    : CustomModule | Record<string, CustomModule>;
+
   private logger: Logger;
 
-  constructor(transport: Transport, logger: Logger = defaultLogger) {
+  constructor(
+    transport: Transport,
+    {
+      logger = defaultLogger,
+      getCustomModule,
+    }: {
+      logger?: Logger;
+      getCustomModule?: CustomGetter;
+    } = {},
```

Comment:
To avoid these inline typings, I would have added something like

```
type ConstructorOptions<CustomGetter> = {
  logger?: Logger;
  getCustomModule?: CustomGetter;
};
```

Outside the class definition, and then used it like this:

```
constructor(
    transport: Transport,
    {
      logger = defaultLogger,
      getCustomModule,
    }: ConstructorOptions<CustomGetter> = {},
  ) {
  
  ```

Reply:
You cannot because CustomGetter is a Generic